sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,551 @@
+                    )
+
+            if train_preds[i] == self._class_dictionary[y[i]]:
+                X_oc.append(train_probas[i])
+
+        one_class_classifier = None
+        if len(X_oc) > 1:
+            cv_size = min(len(X_oc), 10)
+            one_class_classifier = _clone_estimator(
+                self.one_class_classifier, random_state=rs
+            )
+            gs = GridSearchCV(
+                estimator=one_class_classifier,
+                param_grid=self.one_class_param_grid,
+                scoring="accuracy",
+                cv=cv_size,
+            )
+            gs.fit(X_oc, np.ones(len(X_oc)))
+            one_class_classifier = gs.best_estimator_
+
+        return estimator, one_class_classifier, train_probas, train_preds
+
+    def _predict_proba_for_estimator(self, X, i):
+        rs = 255 if self.random_state == 0 else self.random_state
+        rs = None if self.random_state is None else rs * 37 * (i + 1)
+        rng = check_random_state(rs)
+
+        probas = self._estimators[i].predict_proba(
+            X[:, :, : self._classification_points[i]]
+        )
+        preds = [int(rng.choice(np.flatnonzero(prob == prob.max()))) for prob in probas]
+
+        # create data set for the one class classifier using predicted probas with the
+        # minimum difference to the predicted probability
+        X_oc = []
+        if self._one_class_classifiers[i] is not None:
+            X_oc = self._generate_one_class_features(X, preds, probas)
+
+        return X_oc, probas, preds
+
+    def _generate_one_class_features(self, X, preds, probas):
+        # create data set for the one class classifier using predicted probas with the
+        # minimum difference to the predicted probability
+        X_oc = np.hstack((probas, np.ones((len(X), 1))))
+        for i in range(len(X)):
+            for n in range(self.n_classes_):
+                if n != preds[i]:
+                    X_oc[i][self.n_classes_] = min(
+                        X_oc[i][self.n_classes_], X_oc[i][preds[i]] - X_oc[i][n]
+                    )
+        return X_oc
+
+    def _predict_n_timestamps(self, preds, X_oc, consecutive_predictions, n_timestamps):
+        n_instances = len(preds[0])
+
+        # a List containing the state info for case, edited at each time stamp.
+        # contains 1. the index of the time stamp, 2. the number of consecutive
+        # positive decisions made, and 3. the prediction made
+        state_info = [(0, 0, 0) for _ in range(n_instances)]
+        # stores whether we have made a final decision on a prediction, if true
+        # state info wont be edited in later time stamps
+        finished = [False for _ in range(n_instances)]
+
+        for i in range(n_timestamps):
+            if i == len(self._classification_points) - 1:
+                decisions = [True for _ in range(n_instances)]
+            elif self._one_class_classifiers[i] is not None:
```

Comment:
I am not sure, if I understand correctly:  The `predictions` are still made in case of `finished == True` but the `state_info` is not updated anymore.

But, we might as well stop predictions?

Thus, shouldn`t it be?

```
                offsets = np.argwhere(finished==False).flatten()
                decisions_subset = self._one_class_classifiers[i].predict(X_oc[i][offsets]) == 1
                decisions = np.ones(n_instances, dtype=bool)
                decisions[offsets] = decisions_subset
```

This will only do predictions for offsets that are not finished.

Reply:
Yeah, if that works it's better a way of going about it.

Similar could be done for the actual predictions in predict_proba, but unsure if we would be able to reuse the method from _fit.